src/type_object.rs

```diff
@@ -1,46 +1,49 @@
 // Copyright (c) 2017-present PyO3 Project and Contributors
 //! Python type object information
 
-use crate::instance::Py;
 use crate::pyclass::{initialize_type_object, PyClass};
 use crate::pyclass_init::PyObjectInit;
 use crate::types::{PyAny, PyType};
-use crate::{ffi, AsPyPointer, Python};
+use crate::{ffi, AsPyPointer, Py, Python};
 use std::cell::UnsafeCell;
 use std::ptr::NonNull;
 use std::sync::atomic::{AtomicBool, Ordering};
 
-/// `T: PyObjectLayout<U>` represents that `T` is a concrete representaion of `U` in Python heap.
-/// E.g., `PyClassShell` is a concrete representaion of all `pyclass`es, and `ffi::PyObject`
+/// `T: PyLayout<U>` represents that `T` is a concrete representaion of `U` in Python heap.
+/// E.g., `PyCell` is a concrete representaion of all `pyclass`es, and `ffi::PyObject`
 /// is of `PyAny`.
 ///
 /// This trait is intended to be used internally.
-pub trait PyObjectLayout<T: PyTypeInfo> {
+pub unsafe trait PyLayout<T: PyTypeInfo> {
     const IS_NATIVE_TYPE: bool = true;
-
-    fn get_super_or(&mut self) -> Option<&mut <T::BaseType as PyTypeInfo>::ConcreteLayout> {
+    fn get_super_or(&mut self) -> Option<&mut T::BaseLayout> {
```

Comment:
I'd like to suggest just calling this function `super` - I don't think `get` or `or` add anything here.

Reply:
`super` is keyword in Rust so we need some prefix or suffix

Reply:
Ah shoot, good point. How about just `get_super` then?

Reply:
:+1: